server/reload.go

```diff
@@ -57,24 +90,109 @@ func (t *tlsOption) Apply(server *Server) {
 		message = "enabled"
 	}
 	server.generateServerInfoJSON()
+	server.mu.Unlock()
 	server.Noticef("Reloaded: tls = %s", message)
 }
 
+// tlsTimeoutOption implements the option interface for the tls `timeout`
+// setting.
+type tlsTimeoutOption struct {
+	noopOption
+	newValue float64
+}
+
+// Apply is a no-op because the timeout will be reloaded after options are
+// applied.
+func (t *tlsTimeoutOption) Apply(server *Server) {
+	server.Noticef("Reloaded: tls timeout = %v", t.newValue)
+}
+
+// authOption is a base struct that provides default option behaviors.
+type authOption struct{}
+
+func (o authOption) IsLoggingChange() bool {
+	return false
+}
+
+func (o authOption) IsAuthChange() bool {
+	return true
+}
+
+// usernameOption implements the option interface for the `username` setting.
+type usernameOption struct {
+	authOption
+}
+
+// Apply is a no-op because authorization will be reloaded after options are
+// applied.
+func (u *usernameOption) Apply(server *Server) {
+	server.Noticef("Reloaded: authorization username")
+}
+
+// passwordOption implements the option interface for the `password` setting.
+type passwordOption struct {
+	authOption
+}
+
+// Apply is a no-op because authorization will be reloaded after options are
+// applied.
+func (p *passwordOption) Apply(server *Server) {
+	server.Noticef("Reloaded: authorization password")
+}
+
+// authorizationOption implements the option interface for the `token`
+// authorization setting.
+type authorizationOption struct {
+	authOption
+}
+
+// Apply is a no-op because authorization will be reloaded after options are
+// applied.
+func (a *authorizationOption) Apply(server *Server) {
+	server.Noticef("Reloaded: authorization token")
+}
+
+// authTimeoutOption implements the option interface for the authorization
+// `timeout` setting.
+type authTimeoutOption struct {
+	noopOption
```

Comment:
nitpick:  Can you add a comment why not `authOption`? (e.g. // no-op; will be reloaded with options,)